integrations/sklearn/neptune_sklearn/__init__.py

```diff
@@ -0,0 +1,1190 @@
+
+        .. code:: python3
+
+            import neptune.new.integrations.sklearn as npt_utils
+
+            rfc = RandomForestClassifier()
+            rfc.fit(X_train, y_train)
+
+            exp = neptune.init(project='my_workspace/my_project')
+            exp['summary/random_forest'] = npt_utils.create_classifier_summary(rfc, X_train, X_test, y_train, y_test)
+    """
+    assert is_classifier(classifier), 'classifier should be sklearn classifier.'
+
+    cls_summary = dict()
+
+    cls_summary['all_params'] = get_estimator_params(classifier)
+    cls_summary['pickled_model'] = get_pickled_model(classifier)
+
+    y_pred = classifier.predict(X_test)
+
+    cls_summary['test'] = {'preds': get_test_preds(classifier, X_test, y_test, y_pred=y_pred, nrows=nrows),
+                           'preds_proba': get_test_preds_proba(classifier, X_test, nrows=nrows),
+                           'scores': get_scores(classifier, X_test, y_test, y_pred=y_pred)}
+
+    if log_charts:
+        cls_summary['diagnostics_charts'] = {
+            'classification_report': create_classification_report_chart(classifier, X_train, X_test, y_train, y_test),
+            'confusion_matrix': create_confusion_matrix_chart(classifier, X_train, X_test, y_train, y_test),
+            'ROC_AUC': create_roc_auc_chart(classifier, X_train, X_test, y_train, y_test),
+            'precision_recall': create_precision_recall_chart(classifier, X_test, y_test),
+            'class_prediction_error': create_class_prediction_error_chart(classifier, X_train, X_test, y_train, y_test)}
+
+    return cls_summary
+
+
+def create_kmeans_summary(model, X, nrows=1000, **kwargs):
+    """Create sklearn kmeans summary.
+
+    This method fit KMeans model to data and logs:
+
+    * all kmeans parameters,
+    * cluster labels,
+    * clustering visualizations: KMeans elbow chart and silhouette coefficients chart.
+
+    Returned ``dict`` can be assigned to the experiment's namespace defined by the user (see example below).
+
+    Tip:
+        Check Sklearn-Neptune integration
+        `documentation <https://docs-beta.neptune.ai/essentials/integrations/machine-learning-frameworks/sklearn>`_
+        for the full example.
+
+    Args:
+        model (:obj:`KMeans`):
+            | KMeans object.
+        X (:obj:`ndarray`):
+            | Training instances to cluster.
+        nrows (`int`, optional, default is 1000):
+            | Number of rows to log in the cluster labels.
+        kwargs:
+            KMeans parameters.
+
+    Returns:
+        ``dict`` with all summary items.
+
+    Examples:
+        .. code:: python3
+
+            import neptune.new.integrations.sklearn as npt_utils
+
+            km = KMeans(n_init=11, max_iter=270)
+            X, y = make_blobs(n_samples=579, n_features=17, centers=7, random_state=28743)
+
+            exp = neptune.init(project='my_workspace/my_project')
+            exp['summary/kmeans'] = npt_utils.create_kmeans_summary(km, X)
```

Comment:
Information architecture wise shouldn't it be exp['kmeans/summary']?

Reply:
corrected.